pkg/activator/certificate/cache.go

```diff
@@ -40,23 +41,64 @@ import (
 type CertCache struct {
 	secretInformer v1.SecretInformer
 	logger         *zap.SugaredLogger
+	trustConfig    netcfg.Trust
 
-	certificate *tls.Certificate
-	TLSConf     tls.Config
+	certificate   *tls.Certificate
+	ClientTLSConf tls.Config
+	ServerTLSConf tls.Config
 
 	certificatesMux sync.RWMutex
+	caCerts         []*x509.Certificate
+}
+
+func (cr *CertCache) init() {
+	cr.ClientTLSConf.ServerName = certificates.LegacyFakeDnsName
+	cr.ClientTLSConf.MinVersion = tls.VersionTLS13
+	cr.ClientTLSConf.RootCAs = x509.NewCertPool()
+	cr.ClientTLSConf.GetClientCertificate = cr.GetClientCertificate
+
+	cr.ServerTLSConf.MinVersion = tls.VersionTLS12
+	cr.ServerTLSConf.ClientCAs = x509.NewCertPool()
+	cr.ServerTLSConf.GetCertificate = cr.GetCertificate
+	switch cr.trustConfig {
+	case netcfg.TrustIdentity, netcfg.TrustMutual:
+		cr.ServerTLSConf.ClientAuth = tls.RequireAndVerifyClientCert
+		cr.ServerTLSConf.VerifyConnection = func(cs tls.ConnectionState) error {
+			if len(cs.PeerCertificates) == 0 {
+				// Should never happen on a server side
+				cr.logger.Info("mTLS: Failed to verify client connection. Certificate is missing\n")
+				return fmt.Errorf("mTLS: Failed to verify client connection. Certificate is missing")
+			}
+			for _, match := range cs.PeerCertificates[0].DNSNames {
+				// Activator currently supports a single routingId which is the default "0"
+				// Working with other routingId is not yet implemented
+				if match == certificates.DataPlaneRoutingName("0") {
+					return nil
+				}
+				//Until all ingresses work with updated dataplane certificates  - allow also any legacy certificate
+				if match == certificates.LegacyFakeDnsName {
+					return nil
+				}
```

Comment:
I believe is the only one that has tests running in this repo - we can just disable those

Reply:
The path that we chose is to do that in a different PR after the ingresses change and we can test  - we are moving from one working setup to the other progressively.

Reply:
@evankanderson and I have both called out multiple times that we should avoid the extra work of doing this progressively and drop the LegacyDNSName. It seems like a bunch of ingresses have already made the change and are waiting on this PR.

Reply:
I can only speak for net-contour really, but as far as I'm aware, it doesn't configure Contour to send client certs to the activator at all right now. So even if have a case for legacyfakednsname in this PR, the mutual/identity cases will fail since Contour doesn't send client certs.

I would probably make net-contour just use the new names in their client certs from the start.

Is there a reason other ingresses wouldn't do that?

Reply:
> @evankanderson and I have both called out multiple times that we should avoid the extra work of doing this progressively and drop the LegacyDNSName. It seems like a bunch of ingresses have already made the change and are waiting on this PR.

@dprotaso 
1. I removed this addition  
2. @evankanderson also agreed (when we discussed) that **we should be moving progressively from one working solution to another** while adding the new trust levels. 
3. We probably will need the extra work anyhow since once we will have tls trust levels in place (and optimally once we also have e2e tests), we will have cleanups to do.
4. A lot of thought, time, and work is placed into how to do the addition of the new trust levels. I don't see how it is helpful to insist on the stages of the work being less progressive against the better judgment of the community member contributing to the work. This is not helpful in building a healthy community.

Reply:
> I removed this addition

Resolving the github thread helps cue to others that this is complete

> evankanderson also agreed (when we discussed) that we should be moving progressively from one working solution to another while adding the new trust levels.
> A lot of thought, time, and work is placed into how to do the addition of the new trust levels. I don't see how it is helpful to insist on the stages of the work being less progressive against the better judgment of the community member committing the work. This is not helpful in building a healthy community.

The confusion here is that there's tons of discussion that's happened but hasn't been documented in a central place (which should be the feature track). None of the issues in the [project board](https://github.com/orgs/knative/projects/63/views/1) hint towards such a transition.

And @nak3 and I suggested dropping support for minimal completely in the issue here a while back - https://github.com/knative/serving/issues/13968 . So this feedback was provided before you started this work